contrib/NeonVM/cell.c

```diff
@@ -10,54 +10,39 @@
 
 Cell *cell_fromNumber(BID_UINT128 n)
 {
-    Cell *x = malloc(sizeof(Cell));
-    if (x == NULL) {
-        fatal_error("Could not allocate new number object.");
-    }
+    Cell *x = cell_newCell();
     x->type = Number;
     x->number = n;
     return x;
 }
 
 Cell *cell_fromString(const char *s)
 {
-    Cell *x = malloc(sizeof(Cell));
-    if (x == NULL) {
-        fatal_error("Could not allocate new string object.");
-    }
+    Cell *x = cell_newCell();
     x->type = String;
     x->string = _strdup(s);
     return x;
 }
 
 Cell *cell_fromCell(const Cell *c)
 {
-    Cell *x = malloc(sizeof(Cell));
-    if (x == NULL) {
-        fatal_error("Could not allocate new Cell object.");
-    }
     assert(c != NULL);
+
+    Cell *x = cell_newCell();
     x->type = c->type;
     switch (c->type) {
         case Address:
-            memcpy(x, c, sizeof(Cell));
+            cell_copyCell(x, c);
+            //memcpy(x, c, sizeof(Cell));
             //x->address = c->address;
             break;
         case String:
             x->string = _strdup(c->string);
-            x->number = bid128_from_uint32(0);
-            x->address = NULL;
             break;
         case Number:
             x->number = c->number;

```

Comment:
You're no longer setting the other members to NULL here. This may cause a problem if a cell goes from being a string to a number, and keeps the old pointer to the string around. When you go to free that cell, it will call free(c->string) even if the cell is not in fact a string type.